config/src/config/execution_config.rs

```diff
@@ -56,13 +56,41 @@ impl Default for ExecutionConfig {
 impl ExecutionConfig {
     pub fn load(&mut self, root_dir: &RootPath) -> Result<(), Error> {
         if !self.genesis_file_location.as_os_str().is_empty() {
-            let path = root_dir.full_path(&self.genesis_file_location);
-            let mut file = File::open(&path).map_err(|e| Error::IO("genesis".into(), e))?;
+            // Ensure the genesis file exists
+            let genesis_path = root_dir.full_path(&self.genesis_file_location);
+            if !genesis_path.exists() {
+                return Err(Error::Unexpected(format!(
+                    "The genesis file could not be found! Ensure the given path is correct: {:?}",
+                    genesis_path.display()
+                )));
+            }
+
+            // Open the genesis file and read the bytes
+            let mut file = File::open(&genesis_path).map_err(|error| {
+                Error::Unexpected(format!(
```

Comment:
Instead of `Unexpected` error should this be categorized as `IO` error as before?

Reply:
This is me being lazy 😄. `IO` error has a different format. It expects `filename` and `#[source] std::io::Error`. But, this format doesn't match most of these cases and it's not user friendly. This is why I've temporarily added the `Unexpected` error as a catch all. 

If someone wants to go through and refactor all the error types and uses to be more user friendly (there's a lot more than touched by this PR) feel free 😄 But, I won't be doing that now.